OsmAnd/src/net/osmand/plus/activities/MapActivity.java

```diff
@@ -692,9 +692,9 @@ public void onBackPressed() {
 			return;
 		}
 
-		SnapTrackWarningBottomSheet snapTrackWarningBottomSheet = getSnapTrackWarningBottomSheet();
-		if (snapTrackWarningBottomSheet != null) {
-			snapTrackWarningBottomSheet.dismiss();
+		SnapTrackWarningFragment snapTrackWarningFragment = getSnapTrackWarningBottomSheet();
```

Comment:
This method shouldn't be part of MapActivity

Reply:
It's needs for onBackPressed()

Reply:
Should be refactored